x-pack/filebeat/input/cometd/input.go

```diff
@@ -0,0 +1,445 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cometd

```

Comment:
I'm not convinced about keeping all the logic in a single file, I think you have to adjust it based on the SoC principle.

Reply:
Right, we tried following the `gcppubsub` and that's why we kept it all in a single file. We'll try and follow the SoC principle.